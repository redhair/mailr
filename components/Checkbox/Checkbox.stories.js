import React from 'react';
import { storiesOf } from '@storybook/react';
import Checkbox from '.';

storiesOf('Primitives|Checkbox', module).add('default', () => {
  return (
    <>
      <Checkbox checked={true} onChecked={() => {}} name="test" label="Uncheck me" />
      <Checkbox checked={false} onChecked={() => {}} name="test" label="Check me" />
    </>
  );
});
