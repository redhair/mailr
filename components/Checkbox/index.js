import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Text } from '../Typography';

const Wrapper = styled.div`
  position: relative;
  display: block;
  min-height: 1.5rem;
  padding-left: 1.5rem;

  & input[type='checkbox'] {
    box-sizing: border-box;
    padding: 0;
    position: absolute;
    z-index: -1;
    opacity: 0;
  }

  input[type='checkbox']:checked + label {
    & span {
      &:first-child {
        background: ${(props) => props.theme.secondaryColor};
        border-color: ${(props) => props.theme.secondaryColor};
        animation: wave 0.4s ease;

        & svg {
          stroke-dashoffset: 0;
        }

        &:before {
          transform: scale(3.5);
          opacity: 0;
          transition: all 0.6s ease;
        }
      }
    }
  }

  @keyframes wave {
    50% {
      transform: scale(0.9);
    }
  }

  & label {
    margin: auto;
    -webkit-user-select: none;
    user-select: none;
    cursor: pointer;

    & span {
      display: inline-block;
      vertical-align: middle;
      transform: translate3d(0, 0, 0);

      &:first-child {
        position: relative;
        width: 18px;
        height: 18px;
        box-sizing: border-box;
        border-radius: 3px;
        transform: scale(1);
        vertical-align: middle;
        border: 1px solid #9098a9;
        transition: all 0.2s ease;

        & svg {
          position: absolute;
          top: 3px;
          left: 2px;
          fill: none;
          stroke: #ffffff;
          stroke-width: 2;
          stroke-linecap: round;
          stroke-linejoin: round;
          stroke-dasharray: 16px;
          stroke-dashoffset: 16px;
          transition: all 0.3s ease;
          transition-delay: 0.1s;
          transform: translate3d(0, 0, 0);
        }

        &:before {
          content: '';
          width: 100%;
          height: 100%;
          background: ${(props) => props.theme.secondaryColor};
          display: block;
          transform: scale(0);
          opacity: 1;
          border-radius: 50%;
        }
      }

      &:last-child {
        padding-left: 8px;
      }
    }

    &:hover span:first-child {
      border-color: ${(props) => props.theme.secondaryColor};
    }
  }
`;

Checkbox.propTypes = {
  checked: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  withStrikethrough: PropTypes.bool,
  onChecked: PropTypes.func,
};

function Checkbox({ checked, name, label, withStrikethrough, onChecked }) {
  return (
    <Wrapper>
      <input checked={checked} id={name} name={name} type="checkbox" onChange={onChecked} />
      <label htmlFor={name}>
        <span>
          <svg width="12px" height="10px" viewBox="0 0 12 10">
            <polyline points="1.5 6 4.5 9 10.5 1"></polyline>
          </svg>
        </span>
        <Text style={withStrikethrough ? { textDecoration: 'line-through' } : {}}>{label}</Text>
      </label>
    </Wrapper>
  );
}

export default Checkbox;
