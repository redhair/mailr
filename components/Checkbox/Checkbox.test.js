import React from 'react';
import { create } from 'react-test-renderer';
import 'jest-styled-components';
import Checkbox from '.';
import ThemeProvider from '../ThemeProvider';

describe('Button', () => {
  it('Renders a checkbox', () => {
    const tree = create(
      <ThemeProvider>
        <Checkbox checked={true} name="test" label="Check Me" onChecked={() => {}} />
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
