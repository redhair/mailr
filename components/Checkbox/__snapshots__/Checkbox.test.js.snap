// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders a checkbox 1`] = `
.c1 {
  font-family: 'Nunito',sans-serif;
  font-size: 16px;
  line-height: 24px;
}

.c0 {
  position: relative;
  display: block;
  min-height: 1.5rem;
  padding-left: 1.5rem;
}

.c0 input[type='checkbox'] {
  box-sizing: border-box;
  padding: 0;
  position: absolute;
  z-index: -1;
  opacity: 0;
}

.c0 input[type='checkbox']:checked + label span:first-child {
  background: #57c59b;
  border-color: #57c59b;
  -webkit-animation: wave 0.4s ease;
  animation: wave 0.4s ease;
}

.c0 input[type='checkbox']:checked + label span:first-child svg {
  stroke-dashoffset: 0;
}

.c0 input[type='checkbox']:checked + label span:first-child:before {
  -webkit-transform: scale(3.5);
  -ms-transform: scale(3.5);
  transform: scale(3.5);
  opacity: 0;
  -webkit-transition: all 0.6s ease;
  transition: all 0.6s ease;
}

.c0 label {
  margin: auto;
  -webkit-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c0 label span {
  display: inline-block;
  vertical-align: middle;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
}

.c0 label span:first-child {
  position: relative;
  width: 18px;
  height: 18px;
  box-sizing: border-box;
  border-radius: 3px;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  vertical-align: middle;
  border: 1px solid #9098a9;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c0 label span:first-child svg {
  position: absolute;
  top: 3px;
  left: 2px;
  fill: none;
  stroke: #ffffff;
  stroke-width: 2;
  stroke-linecap: round;
  stroke-linejoin: round;
  stroke-dasharray: 16px;
  stroke-dashoffset: 16px;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-transition-delay: 0.1s;
  transition-delay: 0.1s;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
}

.c0 label span:first-child:before {
  content: '';
  width: 100%;
  height: 100%;
  background: #57c59b;
  display: block;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  opacity: 1;
  border-radius: 50%;
}

.c0 label span:last-child {
  padding-left: 8px;
}

.c0 label:hover span:first-child {
  border-color: #57c59b;
}

<div
  className="c0"
>
  <input
    checked={true}
    id="test"
    name="test"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="test"
  >
    <span>
      <svg
        height="10px"
        viewBox="0 0 12 10"
        width="12px"
      >
        <polyline
          points="1.5 6 4.5 9 10.5 1"
        />
      </svg>
    </span>
    <span
      className="c1"
      style={Object {}}
    >
      Check Me
    </span>
  </label>
</div>
`;
